angular/src/lib/drawer/drawer.component.ts

```diff
@@ -0,0 +1,64 @@
+ /** @component drawer */
+import { Component, Input, Output, EventEmitter, HostBinding, ElementRef } from '@angular/core';
+
+export type DrawerOpenFromDirection = 'right' | 'top' | 'bottom';
+export type DrawerSize = 'medium' | 'large';
+
+@Component({
+  selector: 'md-drawer',
+  template: `
+    <ng-content></ng-content>
+    <md-icon
+      className="md-drawer__close"
+      name="cancel_14"
+      ariaLabel="{{closeAriaLabel}}"
+      (click)="close()" >
+    </md-icon>
+  `,
+  styles: []
+})
+export class DrawerComponent {
+
+  private _visible: Boolean = false;
+
+  /** @prop css class names | '' | '' */
+  @Input() public class: string = '';
+
+  @HostBinding('class') get className(): string {
+    return `${this.class ? this.class : '' } `
+      + 'md-drawer '
+      + `${'md-drawer__' + this.openFrom} `
```

Comment:
In terms of BEMs - both openFrom and size seem to be their variations (--) of md-drawer. The additional styling could be done in the structure of the css. @pauljeter